src/App.js

```diff
@@ -1,61 +1,103 @@
-import React from 'react';
-
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
-
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-          autoFocus=""
+import React from "react";
+
+import ToDoItem from "./ToDoItem";
+import Header from "./Header";
+import Footer from "./Footer";
+
+class App extends React.Component {
+  state = {
+    toDoItems: [],
+    selectedFilter: "all"
+  };
+
+  saveToLocalSetState = state => {
+    this.setState({ ...state }, () => {
+      this.saveDataToLocalStorage();
+    });
+  };
+
+  componentDidMount() {
+    if (localStorage.getItem("toDoItems") !== null) {
+      this.saveToLocalSetState({
+        toDoItems: JSON.parse(localStorage.getItem("toDoItems")),
+        selectedFilter: localStorage.getItem("selectedFilter")
+      });
+    }
+  }
+
+  saveDataToLocalStorage() {
+    localStorage.setItem("toDoItems", JSON.stringify(this.state.toDoItems));
+    localStorage.setItem("selectedFilter", this.state.selectedFilter);
+  }
+
+  addNewItem = newItem => {
+    this.saveToLocalSetState({
+      toDoItems: [...this.state.toDoItems, newItem]
+    });
+  };
+
+  toggleItem = id => {
+    const toDoArray = [...this.state.toDoItems];
+    const index = toDoArray.findIndex(item => item.id === id);
+    toDoArray[index].isCompleted = !toDoArray[index].isCompleted;
+    this.saveToLocalSetState({
+      toDoItems: toDoArray
+    });
+  };
+
+  deleteToDoItem = id => {
+    const toDoItems = [...this.state.toDoItems].filter(toDo => toDo.id !== id);
+    this.saveToLocalSetState({
+      toDoItems
+    });
+  };
+
+  setSelectedFiler = selectedFilter => {
+    this.saveToLocalSetState({
+      selectedFilter
+    });
+  };
+
+  generateFilteredDataSouce = () => {
+    const { selectedFilter, toDoItems } = this.state;
+    let res = [];
+    switch (selectedFilter) {
+      default:
```

Comment:
`default` should be last item in `switch` constructions.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch